pkg/cloudprovider/providers/openstack/openstack_routes.go

```diff
@@ -246,7 +247,7 @@ func (r *Routes) DeleteRoute(clusterName string, route *cloudprovider.Route) err
 	}
 	defer onFailure.Call(unwind)
 
-	port, err := getPortByIP(r.network, addr)
+	port, err := getPortByIP(r.compute, r.network, route.TargetNode, addr)
```

Comment:
`updateAllowedAddressPairs` only actually needs the port ID .. so we should change that function to accept port ID (as a string) directly, and then adjust the code here to not bother fetching the full port object.